source/ClientRequest/AMDSClientContinuousDataRequest.cpp

```diff
@@ -57,13 +69,30 @@ AMDSClientContinuousDataRequest& AMDSClientContinuousDataRequest::operator =(con
 	return (*this);
 }
 
+void AMDSClientContinuousDataRequest::setSocketKey(const QString &socketKey)
+{
+	AMDSClientDataRequest::setSocketKey(socketKey);
+
+	foreach (QString bufferName, bufferNames()) {
+		AMDSClientContinuousDataRequest * dataRequest = bufferDataRequestList_.value(bufferName);
+		if (dataRequest)
+			dataRequest->setSocketKey(socketKey);
+	}
+}
+
+void AMDSClientContinuousDataRequest::setBufferNames(const QStringList &names) {
+	bufferNameList_.clear();
```

Comment:
To ensure that you don't accidentally forget to clear lists before appending to them, it would be easier (and less error prone) to use `bufferNameList_ = names;`.